vis.py

```diff
@@ -156,6 +148,81 @@ def audio_data_uri(audio_path, sample_rate = None, audio_backend = 'scipy', audi
 		
 	return data_uri(audio_format = audio_format, audio_bytes = audio_bytes)
 
+
 def fmt_audio(audio_path, channel = 0):
 	return f'<audio id="audio{channel}" style="width:100%" controls src="{audio_data_uri(audio_path)}"></audio>\n'
 
+
+def viz_dataset(ref_path, hyp_path, html_path, debug_audio):
+	assert ref_path is not None or hyp_path is not None
+
+	data_files = []
+	if ref_path is not None:
+		data_files.append(('ref', ref_path))
+	if hyp_path is not None:
+		data_files.append(('hyp', hyp_path))
+
+	examples = None
+	for hypref, path in data_files:
+		buffer = dict()
+		with open(path) as data_file:
+			for line in data_file:
+				example = json.loads(line)
+				if examples is not None and example['audio_path'] not in examples:
+					continue
+				markup = []
+				for speaker in range(1, len(example['markup'])):
+					markup.extend(dict(begin = t['begin'], end = t['end'], speaker = speaker) for t in example['markup'][speaker])
+				markup = sorted(markup, key = lambda x: x['end'])
+				if examples is not None:
+					buffer[example['audio_path']] = examples[example['audio_path']]
+					buffer[example['audio_path']][hypref] = markup
+				else:
+					example.pop('markup')
+					buffer[example['audio_path']] = example
+					buffer[hypref] = markup
+		examples = buffer
+	examples = sorted(examples.values(), key = lambda x: x['audio_name'])
+
+	diarization(examples, html_path, debug_audio)
+
+
+def viz_metrics(metrics_path, html_path, debug_audio):
```

Comment:
if you dislike calling it `metrics()` for name clashes or other reasons, let's keep it `vis_metrics`, `viz` would be needed for name clashes, this seems not the case with these methods